.github/workflows/scripts/check-help-wanted.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+
+set -e
+
+PR_URL="$1"
+
+if [ -z "$PR_URL" ]; then
+    echo "Usage: $0 <pr_url>"
+    exit 1
+fi
+
+# Extract PR number from URL for logging
+PRNUM=$(basename "$PR_URL")
+
+# Get PR details including closing issues references
+PR_DATA=$(gh pr view "$PR_URL" --json closingIssuesReferences)
+
+# Extract closing issues references
+CLOSING_ISSUES=$(echo "$PR_DATA" | jq -r '.closingIssuesReferences[]?.number // empty')
+
+if [ -z "$CLOSING_ISSUES" ]; then
+    echo "No closing issues found for PR #$PRNUM"
+    exit 0
+fi
+
+# Check each closing issue for help-wanted label
+ISSUES_WITHOUT_HELP_WANTED=()
+
+for issue_num in $CLOSING_ISSUES; do
+    echo "Checking issue #$issue_num for help-wanted label..."
+
+    # Get issue labels
+    LABELS=$(gh issue view "$issue_num" --json labels --jq '.labels[].name')
+
+    # Check if help-wanted label exists
+    if ! echo "$LABELS" | grep -q "help-wanted"; then
+        ISSUES_WITHOUT_HELP_WANTED+=("$issue_num")
+        echo "Issue #$issue_num does not have help-wanted label"
+    else
+        echo "Issue #$issue_num has help-wanted label"
+    fi
+done
+
+# If we found issues without help-wanted label, post a comment
+if [ ${#ISSUES_WITHOUT_HELP_WANTED[@]} -gt 0 ]; then
+    echo "Found ${#ISSUES_WITHOUT_HELP_WANTED[@]} issues without help-wanted label"
+
+    # Build issue list for comment
+    ISSUE_LIST=""
+    for issue_num in "${ISSUES_WITHOUT_HELP_WANTED[@]}"; do
+        ISSUE_LIST="$ISSUE_LIST- #$issue_num"$'\n'
+    done
+
+    # Create comment message
+    COMMENT_MESSAGE="Thank you for your pull request! 🎉
+
+This PR appears to fix the following issues that are not labeled with \`help-wanted\`:
+
+$ISSUE_LIST
+As outlined in our [Contributing Guidelines](https://github.com/cli/cli/blob/trunk/.github/CONTRIBUTING.md), we expect that PRs are only created for issues that have been labeled \`help-wanted\`.
+
+While we appreciate your initiative, please note that:
+
+- **PRs for non-\`help-wanted\` issues may not be reviewed immediately** as they might not align with our current priorities
+- **The issue might already be assigned** to a team member or planned for a specific release
+- **We may need to close this PR** if it conflicts with ongoing work or architectural decisions
```

Comment:
Suggestion: I would like this to read more open-ended. We could close a PR for a variety of reasons on a case-by-case basis, but these are just examples.

```suggestion
- **We may need to close this PR**. For example, if it conflicts with ongoing work or architectural decisions
```